modules/network/src/main/java/org/apache/ignite/internal/network/direct/DirectMessageWriter.java

```diff
@@ -488,12 +499,16 @@ public void afterInnerMessageWrite(boolean finished) {
         setBuffer(EMPTY_BYTE_BUFFER);
```

Comment:
It seems that we empty the buffer only for last stream, that's okay?

Reply:
Yes, that's okay, this is exactly what's expected